### PR TITLE
improve .gitignore file; improve ASIC memories Makefiles; fix asic-sim-synth target; move ASIC simulation dir for simulation dir

### PR DESCRIPTION
**Pull request by JDLopes**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -250,16 +249,20 @@ else
...
 endif
 
-asic-sim-synth:
-	make -C $(FIRM_DIR) run BAUD=$(HW_BAUD)
+asic-sim-synth: sim-clean
+	make -C $(FIRM_DIR) run BAUD=$(SIM_BAUD)
+	make -C $(BOOT_DIR) run BAUD=$(SIM_BAUD)
 ifeq ($(shell hostname), $(ASIC_SERVER))
-	make -C $(ASIC_DIR) sim INIT_MEM=$(INIT_MEM) USE_DDR=$(USE_DDR) RUN_DDR=$(RUN_DDR) TEST_LOG=$(TEST_LOG) VCD=$(VCD) BAUD=$(HW_BAUD) SYNTH=1
+	make -C $(HW_DIR)/simulation/ncsim run INIT_MEM=$(INIT_MEM) USE_DDR=$(USE_DDR) RUN_DDR=$(RUN_DDR) TEST_LOG=$(TEST_LOG) VCD=$(VCD) BAUD=$(SIM_BAUD) SYNTH=1
```
Comment (jjts) [inline Makefile:256]: thiis is called SIM_DIR

File: Makefile [hunk 2]
```diff
@@ -250,16 +249,20 @@ else
...
 	ssh $(ASIC_USER)@$(ASIC_SERVER) "if [ ! -d $(REMOTE_ROOT_DIR) ]; then mkdir -p $(REMOTE_ROOT_DIR); fi"
 	rsync -avz --exclude .git $(ROOT_DIR) $(ASIC_USER)@$(ASIC_SERVER):$(REMOTE_ROOT_DIR)
-	ssh -Y -C $(ASIC_USER)@$(ASIC_SERVER) 'cd $(REMOTE_ROOT_DIR); make -C $(ASIC_DIR) sim INIT_MEM=$(INIT_MEM) USE_DDR=$(USE_DDR) RUN_DDR=$(RUN_DDR) TEST_LOG=$(TEST_LOG) VCD=$(VCD) BAUD=$(HW_BAUD) SYNTH=1'
+	ssh $(ASIC_USER)@$(ASIC_SERVER) 'cd $(REMOTE_ROOT_DIR); make -C $(HW_DIR)/simulation/ncsim run INIT_MEM=$(INIT_MEM) USE_DDR=$(USE_DDR) RUN_DDR=$(RUN_DDR) TEST_LOG=$(TEST_LOG) VCD=$(VCD) BAUD=$(SIM_BAUD) SYNTH=1'
+ifeq ($(TEST_LOG),1)
+	scp $(ASIC_USER)@$(ASIC_SERVER):$(REMOTE_ROOT_DIR)/$(HW_DIR)/simulation/ncsim/test.log $(HW_DIR)/simulation/ncsim
+endif
 ifeq ($(VCD),1)
-	scp $(ASIC_USER)@$(ASIC_SERVER):$(REMOTE_ROOT_DIR)/$(ASIC_DIR)/*.vcd $(ASIC_DIR)
+	scp $(ASIC_USER)@$(ASIC_SERVER):$(REMOTE_ROOT_DIR)/$(HW_DIR)/simulation/ncsim/*.vcd $(HW_DIR)/simulation/ncsim
```
Comment (jjts) [inline Makefile:265]: here too

File: Makefile (hunk 1 above)
Comment (JDLopes) [reply Makefile:256]: Not quite. If you run with SIMULATOR=ncsim, yes, it is SIM_DIR. If you run without SIMULATOR defined, it is the default (icarus). So I did as you see in the pull request.

File: Makefile (hunk 2 above)
Comment (JDLopes) [reply Makefile:265]: See the other comment.

**Pull request merged**